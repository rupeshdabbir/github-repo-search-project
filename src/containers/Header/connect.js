// @flow

import { connect } from 'react-redux';

export function mapDispatchToProps(dispatch: Function) {
  return {
    searchForRepos(search: string) {
      dispatch({
        type: 'SEARCH_REQUEST',
        payload: {
          search,
        },
      });
    },
  };
}

export default connect(null, mapDispatchToProps);
