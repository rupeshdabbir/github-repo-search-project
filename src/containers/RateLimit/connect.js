// @flow

import { connect } from 'react-redux';
import get from 'lodash/fp/get';
import format from 'date-fns/format';

function formatDate(timestamp: number): string | void {
  if (!timestamp) {
    return undefined;
  }
  return format(new Date(timestamp * 1000), 'DD-MM-YYYY HH:mm');
}

export function mapStateToProps(state: Object): Object {
  return {
    coreLimit: get('rateLimit.core.limit', state),
    searchLimit: get('rateLimit.search.limit', state),
    searchRemaining: get('rateLimit.search.remaining', state),
    searchReset: formatDate(get('rateLimit.search.reset', state)),
  };
}

export default connect(mapStateToProps);
