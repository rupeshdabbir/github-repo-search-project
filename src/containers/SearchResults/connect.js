// @flow

import { connect } from 'react-redux';
import get from 'lodash/fp/get';

export function mapStateToProps(state: Object): Object {
  return {
    userEntities: get('entities.users', state),
    userIds: get('search.result', state),
    totalResults: get('search.totalResults', state),
    pagination: get('search.pagination', state),
    isPending: get('search.isPending', state),
    totalResponse: get('search.totalResponse.items', state),
  };
}

export default connect(mapStateToProps);
