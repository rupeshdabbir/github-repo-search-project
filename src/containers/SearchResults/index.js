// @flow

import React from 'react';
import { css, StyleSheet } from 'aphrodite/no-important';
import SearchTable from 'components/SearchTable';
import Pagination from 'components/Pagination';
import ResultsMessage from 'components/ResultsMessage';
import ErrorBoundary from 'components/ErrorBoundary';
import Loading from 'components/Loading';
import Container from 'components/Container';
import connect from './connect';

type Props = {
  isPending: boolean,
  pagination: ?Object,
  searchTerm: string,
  totalResults: number,
  userIds: Array<number>,
};

export function SearchResultsContainer(props: Props) {
  const { isPending, pagination, searchTerm, totalResults, userIds, totalResponse } = props;

  console.log('props', props);

  if (!searchTerm) {
    return null;
  }

  if (isPending) {
    return (
      <div className={css(styles.SearchResultsContainer_wrapLoading)}>
        <Loading />
      </div>
    );
  }

  return (
    <Container>
      <div className={css(styles.SearchResultsContainer_item, styles.SearchResultsContainer_wrapResultsMessage)}>
        <ErrorBoundary>
          <ResultsMessage searchTerm={searchTerm} resultsTotal={totalResults} pageTotal={userIds.length} />
        </ErrorBoundary>
      </div>
      <div className={css(styles.SearchResultsContainer_item)}>
        <ErrorBoundary>
          {/* <SearchResults searchTerm={searchTerm} ids={userIds} entities={userEntities} total={totalResults} /> */}
          <SearchTable totalResponse={totalResponse} />
        </ErrorBoundary>
      </div>
      <div className={css(styles.SearchResultsContainer_item)}>
        <ErrorBoundary>
          <Pagination {...pagination} />
        </ErrorBoundary>
      </div>
    </Container>
  );
}

SearchResultsContainer.defaultProps = {
  pagination: {},
  searchTerm: '',
};

const styles = StyleSheet.create({
  SearchResultsContainer_wrapResultsMessage: {
    marginBottom: 30,
  },

  SearchResultsContainer_wrapLoading: {
    marginTop: 25,
  },

  SearchResultsContainer_item: {
    marginTop: 15,
  },
});

export default connect(SearchResultsContainer);
