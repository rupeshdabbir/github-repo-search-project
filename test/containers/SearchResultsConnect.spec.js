import {mapStateToProps} from 'containers/SearchResults/connect';

describe('SearchConnect: mapStateToProps', () => {

  describe('when given the state', () => {
    it('should return relevant properties', () => {
      const state = {
        search: {
          result: [1, 2, 3],
          totalResults: 10,
          pagination: {},
          isPending: true,
        },
        entities: {
          users: {
            1: {id: 1},
            2: {id: 2},
          },
        },
      };
      expect(mapStateToProps(state)).toMatchSnapshot();
    });
  });

});
