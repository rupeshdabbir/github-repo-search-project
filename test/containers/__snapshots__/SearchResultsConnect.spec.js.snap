// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchConnect: mapStateToProps when given the state should return relevant properties 1`] = `
Object {
  "isPending": true,
  "pagination": Object {},
  "totalResponse": undefined,
  "totalResults": 10,
  "userEntities": Object {
    "1": Object {
      "id": 1,
    },
    "2": Object {
      "id": 2,
    },
  },
  "userIds": Array [
    1,
    2,
    3,
  ],
}
`;
